Premium Blocks: Implement the new design

retrofox opened the pull request:
This is the feature branch PR for the design of Premium Blocks.

## 🐞 Issues

* (8P) Premium Blocks: Add Upgrade banner to Placeholder [#44056](https://github.com/Automattic/wp-calypso/issues/44056)
* (5P) Premium Blocks: Add banner below block description in sidebar [#44057](https://github.com/Automattic/wp-calypso/issues/44057
* (8P) Premium Blocks: Replace (paid) label with upgrade indicator [#44048](https://github.com/Automattic/wp-calypso/issues/44048)
* (3P) Premium Blocks: Remove upgrade nudges [#44168](https://github.com/Automattic/wp-calypso/issues/44168)

## 🌱 PRs

* Premium Blocks: Remove Upgrade Nudges [#16598](https://github.com/Automattic/jetpack/pull/16598)
* Premium Blocks: Remove upgrade nudges from frontend [#16614](https://github.com/Automattic/jetpack/pull/16614)
* Premium Blocks: add CSS class to body when Upgrade Nudge is enabled [#16601](https://github.com/Automattic/jetpack/pull/16598
)
* Premium Blocks: Set premium icons for jetpack premium blocks #16602
* Premium Blocks: Add upgrade banner to block inspector [#16603](https://github.com/Automattic/jetpack/pull/16603)
* Premium Blocks: Add Upgrade banner to premium block in the editor canvas [#16618](https://github.com/Automattic/jetpack/pull/16618)
* Premium Blocks: Add premium indicator to donations block icon #16629
* Premium Blocks: Remove paid wording [#1663](https://github.com/Automattic/jetpack/pull/16632)
* Premium Blocks: populate blocks keywords with Premium [#16649](https://github.com/Automattic/jetpack/pull/16649)
* Premium Blocks: Override premium block icons #16696
* Premium Blocks: Handle core/cover block #16688
* Premium Blocks: Handle core/audio block #16717
* Premium Content: load assets using proper hook [#44825](https://github.com/Automattic/wp-calypso/pull/44825)
* Premium Blocks: tracking events [#16882](https://github.com/Automattic/jetpack/pull/16882)
* Premium Blocks: code janitorial 🧽 [#16894](https://github.com/Automattic/jetpack/issues/16894)
* Premium Blocks: set blocks availability [#16898](https://github.com/Automattic/jetpack/pull/16898)

Fixes #


## What are Premium blocks?

All blocks that require a paid plan to increase its functionality.

### Current Premium Blocks list

- Simple Payments (Jetpack)
- Recurring Payments (Jetpack)
- Premium Content (WTP) (partially. It depends on [44825](https://github.com/Automattic/wp-calypso/pull/44825))
- Donations (Jetpack)
- Calendly (Jetpack)
- OpenTable (Jetpack)
- Video (core)
- Audio (core)
- Cover (core)
- WhatsApp (Jetpack)

**IMPORTANT**
* In order to test `core/video`, `core/cover`, `core/audio` and `premium-content/container` you should to apply D46955.
* Premium content requires changes in how the asset files are enqueued. [#44825](https://github.com/Automattic/wp-calypso/pull/44825)

## Design & Behaviour 

### Block Icon

The first sign for the user that the block is premium is its Icon. All premium blocks show a little red circle ⭕️ in their icon.

Inline Inserter | Menu Inserter | 
--------------|---------------
<img src="https://user-images.githubusercontent.com/77539/88173270-ab085780-cbf8-11ea-8acc-91c5f336e3c8.png" width="300px" /> | <img width="360" alt="Screen Shot 2020-07-22 at 8 52 47 AM" src="https://user-images.githubusercontent.com/77539/88173356-cc694380-cbf8-11ea-8548-4d3eef965033.png">
Block Iinspector | BlockToolbar 
<img width="285" alt="Screen Shot 2020-07-22 at 9 18 55 AM" src="https://user-images.githubusercontent.com/77539/88175603-739baa00-cbfc-11ea-94c6-19adfa9301a9.png"> | <img width="298" alt="Screen Shot 2020-07-22 at 9 19 55 AM" src="https://user-images.githubusercontent.com/77539/88175926-ead13e00-cbfc-11ea-8f24-f27a5b228b7b.png">

 ### Upgrade plan Banner

When a premium block is inserted in the editor, it immediately shows the upgrade banner above it (with the exception of `core/cover` block). When the block is not selected, the banner disappears. Also, the banner is shown in the block-inspector/sidebar of the editor.

Editor Canvas | Block Inspector
--------------|---------------
<img src="https://user-images.githubusercontent.com/77539/88176458-b01bd580-cbfd-11ea-92e7-6f625ae5b7dc.png" width="600px" /> | <img src="https://user-images.githubusercontent.com/77539/88178403-b790ae00-cc00-11ea-885a-f02b1c54bb57.png" width="300px" />


#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

* Extend all blocks that should be treated as premium blocks.

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

no

## Testing instructions:

### Testing in dotcom

* Apply these changes to your SB (D47060-code)
* Sandbox a dotcom Simple site
* Go to post edit
* Add a paid block. Use the `premium` term in the blocks selector to get all premium blocks. Confirm you see the premium icon for these blocks.

<img src="https://user-images.githubusercontent.com/77539/88184307-e90d7780-cc08-11ea-961f-f890c80a7780.png" width="300px" />

* Confirm you see the new banner to upgrade the plan, as well as the premium icon.

<img src="https://user-images.githubusercontent.com/77539/88184449-19edac80-cc09-11ea-9955-a6c7ccba56cb.png" width="500px" />

* Confirm that clicking on the Upgrade button, it brings you to the checkout page.

### Upgrade plan

Check that the Upgrade button brings to the user to the proper plan. The upgrade plan is not the same for all blocks. Some blocks require a business plan, personal plan, etc.

### Saving post

<img src="https://user-images.githubusercontent.com/77539/90296382-afbfd480-de61-11ea-880e-25fcd8bc2227.gif" width="500px" />

This PR also handles saving post when the user clicks on the Upgrade button. Check that:

* If the post doesn't have changes, clicking on the Upgrade button leads to the checkout page immediately. 

No changes:
<img src="https://user-images.githubusercontent.com/77539/87611690-0c9f6200-c6df-11ea-944b-72b83d1a43b2.png" width="300px" />

* If there are unsaved changes, confirm that when the user clicks on the Upgrade button, the app saves the post before to perform the redirect. Confirm that it doesn't show the alert modal:

<img src="https://user-images.githubusercontent.com/77539/88185573-77cec400-cc0a-11ea-8d86-1a54b706b5ef.png" width="400px" />

Unsaved changes:
<img src="https://user-images.githubusercontent.com/77539/87611843-728be980-c6df-11ea-8d6a-afd2899f195a.png" width="300px" />

After clicking on the Upgrade button:

<img src="https://user-images.githubusercontent.com/77539/87611935-9e0ed400-c6df-11ea-8f1a-b1201d617c92.png" width="300px" />

In general, confirm that the Upgrade Nudge is not shown anymore

<img src="https://user-images.githubusercontent.com/77539/88185432-4fdf6080-cc0a-11ea-96c3-8704416deb44.png" width="500px" />

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* Implement new design of Premium Blocks.

retrofox committed:
Premium Blocks: Remove Upgrade Nudges (#16598)

* register-jetpack-block: do not wrap premium blocks

* videopress: do not wrap with paid core/video block

retrofox committed:
Premium Blocks: add CSS class to body when Upgrade Nudge is enabled (#16601)

* expose enable_upgrade_nudge property to the client

* plan-utils: add isUpgradeNudgeEnabled() helper fn

* paid-blocks: add custom class to doc body

* extensions: add premium-blocks lib

retrofox committed:
PaidBlocks: Add Paid Icon Indicator (#16556) (#16602)

* Add paid block symbol

* Add paid symbol to paid block icons

* Add enable_upgrade_nudge to the Jetpack_Editor_Initial_State

* Added upgrade nudge body class

* Paid Blocks: dot indicator:
- remove dot indicator from placeholder icon
- delete not used file
- move isUpgradeNudgeEnabled function to  plan-utils

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>
Co-authored-by: Harris Papazolgou <hrrsppzgl@gmail.com>

Co-authored-by: Jerry Jones <jones.jeremydavid@gmail.com>
Co-authored-by: cpap <papazoglou.charalampos@gmail.com>
Co-authored-by: Harris Papazolgou <hrrsppzgl@gmail.com>

retrofox committed:
PaidBlocks: Add Paid Icon Indicator (#16556) (#16602)

* Add paid block symbol

* Add paid symbol to paid block icons

* Add enable_upgrade_nudge to the Jetpack_Editor_Initial_State

* Added upgrade nudge body class

* Paid Blocks: dot indicator:
- remove dot indicator from placeholder icon
- delete not used file
- move isUpgradeNudgeEnabled function to  plan-utils

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>
Co-authored-by: Harris Papazolgou <hrrsppzgl@gmail.com>

Co-authored-by: Jerry Jones <jones.jeremydavid@gmail.com>
Co-authored-by: cpap <papazoglou.charalampos@gmail.com>
Co-authored-by: Harris Papazolgou <hrrsppzgl@gmail.com>

cpapazoglou committed:
Paid Blocks: remove the upgrade nudge banner from frontend (#16614)

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

retrofox committed:
Premium Blocks: Add upgrade banner to block inspector (#16603)

* refact -> create getUpgradeUrl() util helper

* premium-block: add upgrade banner component

* regiter-jetpack-blocks: export requiresPaidPlan fn

* plan-utils: add isUpgradable() helper fn

* premium-blocks: add block edit component

* premium-block: extend block edit for premium

* premium-blocks: fix rebasing issues

* upgrade-banner: replace HOC by hooks

* plan-utils: minor jsdoc improvements

* upgrade-banner: render button with a valid URL

retrofox committed:
Premium Blocks: Add upgrade banner to block inspector (#16603)

* refact -> create getUpgradeUrl() util helper

* premium-block: add upgrade banner component

* regiter-jetpack-blocks: export requiresPaidPlan fn

* plan-utils: add isUpgradable() helper fn

* premium-blocks: add block edit component

* premium-block: extend block edit for premium

* premium-blocks: fix rebasing issues

* upgrade-banner: replace HOC by hooks

* plan-utils: minor jsdoc improvements

* upgrade-banner: render button with a valid URL

retrofox committed:
Premium Blocks: Add Upgrade banner to premium block in the editor canvas (#16618)

* premium-blocks: add HOC to add banner in editor

* premium-blocks: adjust styles for banner in editor

* premium-blocks: extend BlockList for Premium

* premium-blocks: minor improvements

* upgrade-banner: set margin rule stronger

retrofox committed:
premium-blocks: do not hardcode core/video block

cpapazoglou committed:
Paid Blocks: add dot indicator to donations block (#16629)

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

cpapazoglou committed:
Paid Blocks: add dot indicator to donations block (#16629)

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

cpapazoglou committed:
Paid Blocks: remove paid wording (#16632)

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

retrofox committed:
premium-blocks: populate keywords with premium key (#16649)

cpapazoglou committed:
Override premium block icons (#16696)

* Paid Blocks: override icons

* Paid Blocks: icon indicator:
- remove paid icons from hardcoded premium blocks
- fix the the helper function

* Paid Blocks: fixes typo

* Update extensions/shared/premium-blocks/render-premium-icon.js

Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>
Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

cpapazoglou committed:
Override premium block icons (#16696)

* Paid Blocks: override icons

* Paid Blocks: icon indicator:
- remove paid icons from hardcoded premium blocks
- fix the the helper function

* Paid Blocks: fixes typo

* Update extensions/shared/premium-blocks/render-premium-icon.js

Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>
Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

retrofox committed:
Premium Blocks: Handle core/cover block (#16688)

* premium-blocks: add/export context and provider

* upgrade-banner: allow set visibility via context

* plan-utils: remove core/cover exception

* cover: hide upgrade banner by default

* cover: handle banner with media placeholder

* plan-utils: add isStillUsableWithFreePlan helper

* upgrade-banner: handle dual mode

* cover: remove BlockEdit filter/hook

* upgrade-banner: check plan selector before select

* cover: use premium instead cover context

* cover: simplify hooking filters

* plan-utils: jsdoc

* upgrade-banner: set CSS class dinamycally

* cover: fix placeholder styles

* cover: check if cover is upgradable. Clean.

* upgrade-banner: do bot forget to add the plan store

* upgrade-banner: tidy code

retrofox committed:
simple-payments: remove paid symbol from icon

retrofox committed:
simple-payments: remove paid symbol from icon

retrofox committed:
premium-blocks: use @automattic/color-studio

retrofox committed:
premium-blocks: use @automattic/color-studio

obenland commented on extensions/shared/premium-blocks/index.js:
```diff
@@ -0,0 +1,50 @@
+
+		// Extend BlockEdit function.
+		settings.edit = premiumBlockEdit( settings.edit );
+		settings.icon = renderPremiumIcon( settings.icon, name );
```
Can the `name` parameter be removed? It doesn't look like `renderPremiumIcon()` expects it.

cpapazoglou replied:
Sure, I just forgot to remove it https://github.com/Automattic/jetpack/pull/16611/commits/6730cd0471e89a555978c4972560dd526eeda472

obenland commented on extensions/shared/premium-blocks/render-premium-icon.js:
```diff
@@ -0,0 +1,15 @@
+		icon.props.children = [ icon.props.children, <PaidSymbol /> ];
+	}
+	return icon;
+};
```
Can we add tests to this function to make sure `icon.src.props.children` and `icon.props.children` are set correctly? Under which circumstance does an icon have its props within `src`?

cpapazoglou replied:
Thanks for pointing this out @obenland 

Some blocks (eg eventbrite) 

https://github.com/Automattic/jetpack/blob/5f22db71ba41a9301bc5015e74f1f406d302afa9/extensions/blocks/eventbrite/index.js#L39-L49

have their icon declared within a `src` so they can also use other properties as `foreground`. 
Added `jsdoc` and refactored the `src` section to use `cloneElement` for the moment.
I will add some simple tests (for that two specific cases) making sure we don't break some functionality in the near future.

obenland replied:
Great, thanks so much!

cpapazoglou replied:
@obenland , added the 2 basic tests
https://github.com/Automattic/jetpack/pull/16611/commits/b17a7cfe4564be0bb8ad53dea4695bc3f930a70b

you may run them with `yarn test-extensions`

obenland committed:
Clone icon element to prperly set children prop.

Avoids a type error caused by trying to overwrite a read-only property.

obenland committed:
Tighten up banner styles.

obenland committed:
Tighten up banner styles.

retrofox committed:
premium-blocks: use select const

Commit:
Premium Icon: remove unused arguement, added docblock

Commit:
Premium Icon: adds basic tests

Commit:
Premium Icon: adds basic tests

retrofox committed:
Premium Blocks: Handle core/audio block (#16717)

* cover: refact -> move helper function to shared

* cover: move media placeholder out of cover

* premium-blocks: handle media placeholder component

* plan-utils: set audio as dual behavior

* premium-block: move media placeholder styles

* premium-blocks: move helper to utils

* premium-blocks: refact plan utils helpers

* premium-blocks: remove blocks/cover folder. Handle audio

* format code

* premum-block: rename replace flow HOC

* premium-blocks: rename hekper fn to isFileOfType

* premium-block: control replace flow in core/audio

retrofox committed:
Premium Blocks: Handle core/audio block (#16717)

* cover: refact -> move helper function to shared

* cover: move media placeholder out of cover

* premium-blocks: handle media placeholder component

* plan-utils: set audio as dual behavior

* premium-block: move media placeholder styles

* premium-blocks: move helper to utils

* premium-blocks: refact plan utils helpers

* premium-blocks: remove blocks/cover folder. Handle audio

* format code

* premum-block: rename replace flow HOC

* premium-blocks: rename hekper fn to isFileOfType

* premium-block: control replace flow in core/audio

retrofox committed:
premium-blocks: tweak some styles

retrofox committed:
plan-utils: donot check site type when upgradable

retrofox committed:
premium-blocks: add README file

retrofox committed:
premium-blocks: add README file

retrofox committed:
upgrade-banner: fix overlaping

retrofox commented on extensions/shared/premium-blocks/editor.scss:
```diff
@@ -51,7 +51,7 @@ $icon-background-color: #fff;
 	position: relative;
 	top: $banner-height - ( $grid-unit-15 / 2 );
-	z-index: 1;
+	z-index: 10;
```
![image](https://user-images.githubusercontent.com/77539/89900794-92b5a800-dbba-11ea-966f-ea9093e8df53.png)

jeryj commented:
I noticed some issues when using `https://wordpress.com/block-editor/post/[site]/[postId]` vs `[site]/wp-admin/post.php?post=[id]&action=edit`.

### Some issues when using the WPcom editor (`https://wordpress.com/block-editor/post/[site]/[postId]`):

**Leave site? alert shows**
<img width="1389" alt="Leave site alert on wordpress.com route" src="https://user-images.githubusercontent.com/967608/89943455-d4a71400-dbe3-11ea-9f2d-0c656aaeda63.png">

**Route doesn't change to a new checkout url**
<img width="1009" alt="Screen Shot 2020-08-11 at 3 05 12 PM" src="https://user-images.githubusercontent.com/967608/89943573-0fa94780-dbe4-11ea-80e0-4b503ac9bace.png">

This ends up being an issue since you can't use the back button in the browser. Whenever I use the back button, it takes me to a site didn't connect error page. I have to reload to get the editor back.

<img width="1015" alt="Screen Shot 2020-08-11 at 3 05 40 PM" src="https://user-images.githubusercontent.com/967608/89943629-2059bd80-dbe4-11ea-9b0b-256fc491ac90.png">

### Issues for both `wp-admin` and `wordpress.com/block-editor/...` routes

If you use the X button, then it keeps the same edit post URL, but you're now on the plans page.  I would prefer it go back to the post editor as that's where you were. It's odd dismissing an overlay like that and now being on a different page entirely that you haven't seen before.

<img width="1023" alt="Screen Shot 2020-08-11 at 3 07 04 PM" src="https://user-images.githubusercontent.com/967608/89943756-539c4c80-dbe4-11ea-9fbe-9acaedd00431.png">

**I was unable to access the Block Editor Upgrade your plan to use this premium block Upgrade button**

<img width="681" alt="Screen Shot 2020-08-11 at 3 31 39 PM" src="https://user-images.githubusercontent.com/967608/89946244-3ec1b800-dbe8-11ea-861a-aa9f45c9789b.png">

This is likely by design in Gutenberg. I'm not sure how you'd go about doing this 😞 Fortunately, you _can_ access the upgrade button from the block settings sidebar.


Other things to mention that I think are addressed in other codebases:
- Premium Content block does not have the dot indicator
- WhatsApp block does not have the dot indicator

### Overall things are working well
- I see the upgrade banner on the ones that do show the dot
- It shows and hides well 👍 
- The old upgrade nudges are gone
- Searching for "Premium" in the block search shows the premium blocks
- From wp-admin based urls (`[site]/wp-admin/post.php?post=[id]&action=edit`) the upgrade flow works well with saving (and never showing the "Leave site?" alert)

This is a huge amount of work overall. It's really impressive how much you've gotten done on this, @retrofox! Huge kudos to you for all that you've done 🙌

jeryj commented on extensions/shared/premium-blocks/upgrade-plan-banner.jsx:
```diff
@@ -0,0 +1,100 @@
+					href={ checkoutUrl } // Only for server-side rendering, since onClick doesn't work there.
+					onClick={ goToCheckoutPage }
+					className="is-primary"
+					label={ buttonText }
+					title={ buttonText }
```
These lines can both be removed. 

The `title` attribute you never need to use on anything :). The `label` ends up applying to an `aria-label`, which is unnecessary since it matches the same text as the button. Just the normal button text `Upgrade` is all we need here.

retrofox replied:
thanks! updating...

jeherve commented on class.jetpack-gutenberg.php:
```diff
@@ -769,6 +769,7 @@ public static function enqueue_block_editor_assets() {
 				'jetpack'          => array(
 					'is_active'                 => Jetpack::is_active(),
 					'is_current_user_connected' => $is_current_user_connected,
+					'enable_upgrade_nudge'      => apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ),
```
```suggestion
					/** This filter is documented in class.jetpack-gutenberg.php */
					'enable_upgrade_nudge'      => apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ),
```

jeherve commented on extensions/shared/premium-blocks/index.js:
```diff
@@ -0,0 +1,66 @@
+const jetpackPremiumBlock = ( settings, name ) => {
+	if ( isUpgradable( name ) ) {
+		// Populate block keywords.
+		settings.keywords = uniq( [ ...settings.keywords, 'premium', __( 'premium' ) ] );
```
```suggestion
		settings.keywords = uniq( [ ...settings.keywords, 'premium', __( 'premium', 'jetpack' ) ] );
```

jeherve commented on extensions/shared/premium-blocks/upgrade-plan-banner.jsx:
```diff
@@ -0,0 +1,100 @@
+	onRedirect,
+	align,
+	className,
+	title = __( 'Premium Block' ),
+	description = __( 'Upgrade your plan to use this premium block' ),
+	buttonText = __( 'Upgrade' ),
```
```suggestion
	title = __( 'Premium Block', 'jetpack' ),
	description = __( 'Upgrade your plan to use this premium block', 'jetpack' ),
	buttonText = __( 'Upgrade', 'jetpack' ),
```

jeherve reviewed (changes requested):
Gave it a try, I really like the new look :) I left some comments below.

********

Overall, I am not sure how I feel about the "Premium" wording. Until now, "Premium" has been associated with the Premium plan. With that change, it seems that "Premium" becomes a synonym for "Paid".

I think it may become problematic / confusing with blocks that are paid, but that do not actually require a Premium Plan.

- We do not have a Personal plan in Jetpack at the moment, but it still exists on WordPress.com, and some blocks like the Payments block only require a Personal plan only, not a Premium Plan.

**Before**

![image](https://user-images.githubusercontent.com/426388/90006021-6c663a00-dc98-11ea-944b-aef559ce88d9.png)

**After**

![image](https://user-images.githubusercontent.com/426388/90006027-6ff9c100-dc98-11ea-9b35-68fda93ddf9e.png)

- In the future, I could imagine us having more blocks, and some of them may require a Business plan for example.

********

I'm seeing some alignment issues and some z-index issues in WP 5.4 / no Gutenberg / Twenty Twenty

![image](https://user-images.githubusercontent.com/426388/90004746-75eea280-dc96-11ea-9ae4-bb0f48958584.png)

********

The banner does not quite work when you use a dark theme (tested using https://github.com/WordPress/gutenberg-starter-theme )

![image](https://user-images.githubusercontent.com/426388/90006423-22318880-dc99-11ea-883f-d05783f1fbf8.png)

********

In the past, the editor banner would remain visible even when the block was not in focus:

![image](https://user-images.githubusercontent.com/426388/90006508-4f7e3680-dc99-11ea-9df1-53bba2b7e4e9.png)

Should we do the same here?

![image](https://user-images.githubusercontent.com/426388/90006545-63c23380-dc99-11ea-8dc5-4eea7a5c0f79.png)

********

~~On WordPress.com, I seem to have lost the video and cover upgrade banners~~
Scratch that, I had forgotten to apply D46955-code.

********

The banner gets added above the cover block when needed, but not in the block sidebar:

![image](https://user-images.githubusercontent.com/426388/90007549-39717580-dc9b-11ea-97ae-3c5751f56228.png)

********

Should we update the look of the banner of the frontend in a follow-up PR (this was for the Pay with PayPal block):

![image](https://user-images.githubusercontent.com/426388/90007796-a71da180-dc9b-11ea-8323-4e892a030568.png)

retrofox committed:
Update class.jetpack-gutenberg.php

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

retrofox committed:
Update extensions/shared/premium-blocks/upgrade-plan-banner.jsx

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

davemart-in commented:
RE: "Premium" vs. "Paid", @amamujee has asked that we:

- Make all payment blocks available on all plans (even the free plan). We've got a card on the Earn team backlog for this already.
- At which point, the only remaining paid block that would not require a premium plan would be the Audio block which simply requires any paid plan ([Support doc](https://wordpress.com/support/wordpress-editor/blocks/audio-block/))

I think I'm okay with keeping this as "Premium blocks" for now with the understanding that any paid block we add moving forward would be gated at the Premium plan.

retrofox committed:
paid-blocks: is the new premium-blocks

retrofox committed:
paid-blocks: is the new premium-blocks

retrofox commented:
> I noticed some issues when using `https://wordpress.com/block-editor/post/[site]/[postId]` vs `[site]/wp-admin/post.php?post=[id]&action=edit`.

Thanks @jeryj for your feedback. Could you check if this behavior also happens with the current implementation? We haven't had changed anything in the checkout process, only refactoring the process of getting the checkout URL.

retrofox commented:
> I was unable to access the Block Editor Upgrade your plan to use this premium block Upgrade button

It's a known issue already handled in other PRs/patches. Keep in mind this block is not registered by Jetpack, that's the reason why you don't see the banner for this one.

retrofox commented:
> Premium Content block does not have the dot indicator 

Dealing with this one https://github.com/Automattic/wp-calypso/pull/44825

retrofox committed:
upgrade-banner: remove unneeded title attr

retrofox committed:
upgrade-banner: remove unneeded title attr

retrofox commented:
> I'm seeing some alignment issues and some z-index issues in WP 5.4 / no Gutenberg / Twenty Twenty

<img src="https://user-images.githubusercontent.com/77539/90028214-c4e50980-dc8f-11ea-8369-80f912f011d1.png" width="600px" />

Is it a Button block? The thing is the block is still usable so overlapping the block toolbar with the banner could be confusing/annoying. Not sure if inverting the overlap (z-index) could bring a solution. I'm tempted to say that it's ok considering that it's a paid block and will be fixed once the site gets a paid plan.

retrofox commented:
> In the past, the editor banner would remain visible even when the block was not in focus:
> Should we do the same here?

No, we shouldn't. This implementation defines showing the banner only when the block is selected.

retrofox commented:
> > I'm seeing some alignment issues and some z-index issues in WP 5.4 / no Gutenberg / Twenty Twenty
> 
> <img alt="" width="600px" src="https://user-images.githubusercontent.com/77539/90028214-c4e50980-dc8f-11ea-8369-80f912f011d1.png">
> 
> ~Is it a Button block?~ The thing is the block is still usable so overlapping the block toolbar with the banner could be confusing/annoying. Not sure if inverting the overlap (z-index) could bring a solution. I'm tempted to say that it's ok considering that it's a paid block and will be fixed once the site gets a paid plan.

@jeherve what block is it?

retrofox commented:
> Should we update the look of the banner of the frontend in a follow-up PR (this was for the Pay with PayPal block):

Actually, we should remove that nudge in the front-end according to the new design for paid blocks.

retrofox committed:
upgrade-banner: improve dark styles

retrofox commented:
> The banner does not quite work when you use a dark theme (tested using https://github.com/WordPress/gutenberg-starter-theme )
Is gutenberg-starter-theme a dark theme? I've tested and checked with `Corel Dark`

Tackled [here](The banner gets added above the cover block when needed, but not in the block sidebar:)
![image](https://user-images.githubusercontent.com/77539/90032252-8ef65400-dc94-11ea-8089-f03c4d2b7eab.png)

retrofox commented:
> > Should we update the look of the banner of the frontend in a follow-up PR (this was for the Pay with PayPal block):
> 
> Actually, we should remove that nudge in the front-end according to the new design for paid blocks.

Was checking if the Upgrade Nudge is rendered in the front-end because we [already addressed](https://github.com/Automattic/jetpack/pull/16611/commits/218df46d770a2c5ad08e35442b4b8434065a5f85) this requirement. Maybe you forgot to apply these changes in your testing site?

retrofox commented:
> The banner gets added above the cover block when needed, but not in the block sidebar:

Actually the Upgrade banner already appears in the Block inspector but below to the block styles card.

<img src="https://user-images.githubusercontent.com/77539/90035649-6b350d00-dc98-11ea-880f-80728d1bc79a.png" width="400px" />
We created [a PR in core](https://github.com/WordPress/gutenberg/pull/23993) to handle this issue properly. In the meanwhile, we see three options:

1) Move the banner using DOM manipulation: 100% unrecommended. In fact, we did it in a previous PR but discarded because it brought too much complexity and the approach will always be fragile.
2) Hide the banner in the Block inspector for this block
3) Do not show styles for this block.

Going to work on the third option since I considered it's easier and makes sense. Feedback is more than welcomed.

retrofox commented:
> Do not show styles for this block.

AFAIK, it removes the styles from Paid blocks registered by Jetpack.

retrofox committed:
paid-blocks: unregister styles for paid blocks

retrofox commented:
Thanks @jeherve @jeryj for your review/feedback. We've addressed all issues so let me move the PR again in RfR state.

Remarkable changes:

✔️ [locate the code in `./extensions/extended-blocks/paid-blocks`](31ef531ca54e9bead4c362056ff564f1b81a3694)
✔️ [remove unneeded title attr](9ae26974ff8286de98584817b510f135c81d6d00)
✔️ [Tweak styles for dark themes](3f9c1e6abe7836f6819afcfd2765a4a69107d0d3)
✔️ [Remove styles for all paid blocks (Not only registered by Jetpack)](bda645b11a250762657e88a95a9dd589fc222502)

retrofox committed:
paid-blocks: remove console.log

jeryj commented on extensions/extended-blocks/paid-blocks/upgrade-plan-banner.jsx:
```diff
@@ -0,0 +1,103 @@
+					href={ checkoutUrl } // Only for server-side rendering, since onClick doesn't work there.
+					onClick={ goToCheckoutPage }
+					className="is-primary"
+					label={ buttonText }
```
This `label` can get removed too, as the button doesn't need an `aria-label` on it.

retrofox replied:
🤦 Removing...

retrofox replied:
Fixed https://github.com/Automattic/jetpack/pull/16611/commits/b073eb211535086aaba604db9c4978625f9e1954

jeryj commented:
> Could you check if this behavior also happens with the current implementation? We haven't had changed anything in the checkout process, only refactoring the process of getting the checkout URL.

Great question, @retrofox!  I just tested on staging (without the PR) using Calendly (paid) and the old upgrade banner. 
- The URL _does_ change to `https://wordpress.com/checkout/[site].wordpress.com/premium?redirect_to=%2Fpost%2F[site].wordpress.com%2F2518%3Fplan_upgraded%3D1#step2`, so the back button still works
- The X button _does not_ return you to the page (same behavior in this PR), so it would not be blocking.

> I was unable to access the Block Editor Upgrade your plan to use this premium block Upgrade button

For this one, I don't mean the premium content block, but the actual Upgrade button. You can't navigate to the Upgrade button on the block editor canvas while only using the keyboard.

retrofox commented:
> > Could you check if this behavior also happens with the current implementation? We haven't had changed anything in the checkout process, only refactoring the process of getting the checkout URL.
> 
> Great question, @retrofox! I just tested on staging (without the PR) using Calendly (paid) and the old upgrade banner.
> 
> * The URL _does_ change to `https://wordpress.com/checkout/[site].wordpress.com/premium?redirect_to=%2Fpost%2F[site].wordpress.com%2F2518%3Fplan_upgraded%3D1#step2`, so the back button still works
> * The X button _does not_ return you to the page (same behavior in this PR), so it would not be blocking.
> 
> > I was unable to access the Block Editor Upgrade your plan to use this premium block Upgrade button
> 
> For this one, I don't mean the premium content block, but the actual Upgrade button. You can't navigate to the Upgrade button on the block editor canvas while only using the keyboard.

We've been testing it, comparing the URL between the banner (this PR) and the Upgrade nudge Url (prod), and they seem to be the same. That's the testing scope for this PR. Just FYI, it's pretty probable we're going to keep improving this feature making focus in the checkout flow.

retrofox committed:
upgrade-banner: remove label prop

retrofox committed:
upgrade-banner: remove label prop

retrofox commented:
Actually there is something wrong in the checkout URL. Nice 👀 , Jerry.

<img src="https://user-images.githubusercontent.com/77539/90063891-62583180-dcc0-11ea-8581-bb1d042dd60d.png" width="600px" />

retrofox committed:
upgrade-banner: add target to upgrade link

retrofox commented:
> Actually there is something wrong in the checkout URL. Nice 👀 , Jerry.
> 
> <img alt="" width="600px" src="https://user-images.githubusercontent.com/77539/90063891-62583180-dcc0-11ea-8581-bb1d042dd60d.png">

addressed [here](https://github.com/Automattic/jetpack/pull/16611/commits/541097e1c564ca98c63154c74219756a442a0848)

getdave commented on extensions/extended-blocks/paid-blocks/README.md:
```diff
@@ -0,0 +1,20 @@
+
+### Setting blocks as paid in Jetpack
+
+You can take a look at the Extension doc, [Paid Blocks](extensions/README.md#paid-blocks) section to know more about this process.
```
```suggestion
Please refer to the extensions documentation on [Paid Blocks](extensions/README.md#paid-blocks) to learn more about this process.
```

getdave commented on extensions/extended-blocks/paid-blocks/README.md:
```diff
@@ -0,0 +1,20 @@
+
+You can take a look at the Extension doc, [Paid Blocks](extensions/README.md#paid-blocks) section to know more about this process.
+
+### Setting blocks as paid not registered by Jetpack
```
```suggestion
### Setting blocks as paid which are not registered by Jetpack
```

getdave commented on extensions/extended-blocks/paid-blocks/README.md:
```diff
@@ -0,0 +1,20 @@
+
+### Setting blocks as paid not registered by Jetpack
+
+Sometimes some blocks need to be defined as paid because some of their features depend on the site plan, for instance, `core/video`. It's a core block.
```
```suggestion
Sometimes blocks which are not themselves registered by Jetpack need to be marked as paid because some of their features depend on a site plan. A good example of this is the `core/video` block which is provided by Gutenberg Core and requires a paid plan in order to enable the uploading of video files.
```

getdave commented on extensions/extended-blocks/paid-blocks/README.md:
```diff
@@ -0,0 +1,20 @@
+
+Sometimes some blocks need to be defined as paid because some of their features depend on the site plan, for instance, `core/video`. It's a core block.
+
+For these cases, set the extension in WordPress.com in the same way that a block registered by Jetpack, also add the dependency to the paid blocks list. Use D46955-code as a reference for how to do it. 
```
```suggestion
For such cases, you will need to mark the extension as "paid" within the WordPress.com codebase (not within Jetpack). You will also need to add the dependency to the paid blocks list. Please refer to D46955-code as a reference for how to achieve this.
```

getdave commented on extensions/extended-blocks/paid-blocks/edit.js:
```diff
@@ -0,0 +1,27 @@
+	return (
+		<Fragment>
+			<InspectorControls>
+				<UpgradePlanBanner description={ null } blockName={ props.name } />
```
Why do we need to mark explicitly as `null` here?

retrofox replied:
Because the banner has defined a description as default. We need to set as `null` to remove it when it's rendered the block inspector.

getdave commented on extensions/extended-blocks/paid-blocks/index.js:
```diff
@@ -0,0 +1,84 @@
+		// Populate block keywords.
+		settings.keywords = uniq( [ ...settings.keywords, 'premium', __( 'premium', 'jetpack' ) ] );
+
+		// Extend BlockEdit function.
```
```suggestion
		// Enhance BlockEdit function with HOC.
```

getdave commented on extensions/extended-blocks/paid-blocks/index.js:
```diff
@@ -0,0 +1,84 @@
+	if ( isUpgradeNudgeEnabled() ) {
+		document.body.classList.add( 'jetpack-enable-upgrade-nudge' );
+
+		// Unregister block styles for all paid blocks.
```
Why do we need to do this? It isn't immediately clear.

retrofox replied:
It's a hack solution to put the banner immediately after the block-info card. Here there [is the comment](https://github.com/Automattic/jetpack/pull/16611#issuecomment-672948812).

getdave commented on extensions/extended-blocks/paid-blocks/index.js:
```diff
@@ -0,0 +1,84 @@
+);
+
+/*
+ * Add the `jetpack-enable-upgrade-nudge` css Class
```
This is describing _what_ we are doing. I'd like the comment to tell me why we are doing it in order that I can understand the context for this code.

retrofox replied:
👍 Let me ask to @jeryj  since if I'm not wrong he did this implementation. I can work on this, too.

getdave reviewed (dismissed):
Thanks for continuing to work on this @retrofox. It isn't a easy task.

That said I found reviewing this PR difficult because of the _size_ of the PR. I assume this is due to it being a feature for reasons of expediency when working in Jetpack repo?

------------------

I tested your work by applying the patch D47060-cod as requested. I used a **Simple site** on a **Free** plan and confirmed I was **sandboxed** and on **Staging**. 

I saw the following results:

> Add a paid block. Use the premium term in the blocks selector to get all premium blocks. Confirm you see the premium icon for these blocks.

Found the following blocks missing the pink dot:

* Premium Content.
* WhatsApp Button.

| Block Picker | Inline Picker
| -------------- | --------- 
| ![Screen Shot 2020-08-13 at 09 41 30](https://user-images.githubusercontent.com/444434/90113447-464ca280-dd49-11ea-946b-179fe4afcebf.png) | ![Screen Shot 2020-08-13 at 09 45 23](https://user-images.githubusercontent.com/444434/90113765-bc510980-dd49-11ea-8038-eadc6fd786f6.png)

> Confirm you see the new banner to upgrade the plan, as well as the premium icon.

All blocks (inc WhatsApp) had the upgrade nudge and the premium icon dot _**except**_ the Premium Content block:

<img width="874" alt="Screenshot 2020-08-13 at 09 48 28" src="https://user-images.githubusercontent.com/444434/90114169-44cfaa00-dd4a-11ea-9a49-15a78dcc2432.png">

> Saving post

None of the required behaviours worked. Clicking on `Upgrade` caused the alert to show and none of the changes were saved.

![Screen Capture on 2020-08-13 at 09-51-56](https://user-images.githubusercontent.com/444434/90114510-b9a2e400-dd4a-11ea-935a-b91772fbd3f7.gif)

---------

As "Saving post" didn't work I didn't proceed with futher testing.

I have a suspicion that the diff is out of date with the changes in Jetpack. The best way to fix is to follow instructions in PCYsg-qiB-p2.

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
Update extensions/extended-blocks/paid-blocks/index.js

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox commented on extensions/extended-blocks/paid-blocks/README.md:
```diff
@@ -0,0 +1,20 @@
+
+This shared extension takes over providing additional behavior to a block it's considered paid.
+
+This condition can depend on a few factors: is it a simple or atomic site? have the have site a paid or free plan? However, from the perspective of paid-blocks implementation, it relies on the business-layer system.
```
> Improve grammar and comprehension.

🙇‍♂️

retrofox commented on extensions/extended-blocks/paid-blocks/README.md:
```diff
@@ -0,0 +1,20 @@
+
+This shared extension takes over providing additional behavior to a block it's considered paid.
+
+This condition can depend on a few factors: is it a simple or atomic site? have the have site a paid or free plan? However, from the perspective of paid-blocks implementation, it relies on the business-layer system.
```
Yes, let me create a card to improve the document and remove this file from here. he documentation is poor because we haven't found a final solution for this issue.

retrofox commented:
> Found the following blocks missing the pink dot:
> 
> * Premium Content.

We need to change how the plugin enqueue the assets files, Here is the WIP PR: https://github.com/Automattic/wp-calypso/pull/44825


> * WhatsApp Button.

Looking 👀 

> > Saving post
> 
> None of the required behaviours worked. Clicking on `Upgrade` caused the alert to show and none of the changes were saved.

Did you test on Calypso or wp-admin?

> I have a suspicion that the diff is out of date with the changes in Jetpack. The best way to fix is to follow instructions in PCYsg-qiB-p2.

I don't think so but it's probably. You never know. I've kept the patch updated constantly, but sometimes just happens. 🤷 Let me take a look.

retrofox commented:
> > * WhatsApp Button.
> 
> Looking 👀

it's funny because the icon variation is applied to the parent block. However, the block shows the icon of the children block.

<img width="302" src="https://user-images.githubusercontent.com/77539/90129344-db966980-dd3e-11ea-9b92-933ac381e38a.png" />
<img width="402" alt="Screen Shot 2020-08-13 at 8 27 53 AM" src="https://user-images.githubusercontent.com/77539/90129399-f832a180-dd3e-11ea-8bb8-5ca0620a99c8.png">

retrofox commented:
> > > * WhatsApp Button.
> > 
> > 
> > Looking 👀
> 
> it's funny because the icon variation is applied to the parent block. However, the block shows the icon of the children block.
> 
> <img alt="" width="302" src="https://user-images.githubusercontent.com/77539/90129344-db966980-dd3e-11ea-9b92-933ac381e38a.png">

It's because we are redefining the icon through the block variations

retrofox committed:
paid-icon: add extend helper function

retrofox committed:
send-a-message: extend default-variation icon

retrofox commented:
we've addressed the WhatsApp case [extending the icon variation](https://github.com/Automattic/jetpack/pull/16611/commits/5b43e5d9998b80a83f597f93b7fa378198faf951) in the block registration process, instead of using a filter.

retrofox committed:
paid-blocks: format code

mreishus commented:
Not sure if fixable - When clicking on the WhatsApp button, the "Upgrade your plan to use this premium block" message is partially covered.  It's because it's two components; if you click to the right of the green button it works ok.
![2020-08-13_14-39](https://user-images.githubusercontent.com/937354/90179592-482b5b80-dd73-11ea-8988-6ad2c584b39b.png)
![2020-08-13_14-41](https://user-images.githubusercontent.com/937354/90179604-4b264c00-dd73-11ea-9f04-6b619b1bbc9a.png)

mreishus commented:
As a free user, `core/cover` allows me to upload an image, but I must upgrade to use a video.  I found the behavior confusing and not clearly communicated (I didn't understand what was happening until @jeryj  told me).

For example, this makes me think I have to pay to use cover, but I don't if I only use an image:
![2020-08-13_14-47](https://user-images.githubusercontent.com/937354/90180114-1f579600-dd74-11ea-9f05-39a067666c60.png)

Most blocks show both a red circle in icon, and an upgrade banner when you first add them, example:
![2020-08-13_14-49](https://user-images.githubusercontent.com/937354/90180239-4ca44400-dd74-11ea-98dc-adf1b641fdd8.png)

However, cover shows only a red circle in the icon, with no upgrade banner:
![2020-08-13_14-50](https://user-images.githubusercontent.com/937354/90180321-680f4f00-dd74-11ea-9a45-1e75cfafbbed.png)

That tells me something is different, but as a user I didn't really understand what.

The upgrade banner only appears when you try to upload a video.
![2020-08-13_14-52](https://user-images.githubusercontent.com/937354/90180473-a0169200-dd74-11ea-8c21-17b2b13b2261.png)

It works fine, but there's not a clear messaging for a block that is premium only when used in certain ways.  It says "Upgrade your plan to use this premium block," which is not entirely correct, since I can use the block as long as I select an image instead of a video.

It doesn't seem that there's an easy fix for this, though, so I'm not sure what to recommend as a solution, or if this should even hold up the release of it.

obenland commented:
Yeah, maybe we could just remove the premium indicator for the Cover block? it would then alert users only when they try to use the premium feature. @davemart-in Do you have thoughts on that?

retrofox commented:
> Not sure if fixable - When clicking on the WhatsApp button, the "Upgrade your plan to use this premium block" message is partially covered. It's because it's two components; if you click to the right of the green button it works ok.
> ![2020-08-13_14-39](https://user-images.githubusercontent.com/937354/90179592-482b5b80-dd73-11ea-8988-6ad2c584b39b.png)
> ![2020-08-13_14-41](https://user-images.githubusercontent.com/937354/90179604-4b264c00-dd73-11ea-9f04-6b619b1bbc9a.png)

Yes, I'm not sure what's the best solution here:

* Elevate (z-index) the banner, but we are going to hide the block toolbar.
* Disable the block in order to don't allow interaction.

If I'm not wrong, we agreed that we should keep the block usable so I guess the current state is the best option. cc @sfougnier

retrofox commented:
> It doesn't seem that there's an easy fix for this, though, so I'm not sure what to recommend as a solution, or if this should even hold up the release of it.

I think it isn't a big deal. These blocks are special in the sense they can be usable with a free plan. Maybe we should reword the banner message?

sfougnier commented:
> If I'm not wrong, we agreed that we should keep the block usable so I guess the current state is the best option. cc @sfougnier

Agreed that the current state is the best option. It's not ideal, but it seems like there isn't a better solution for this. I am however curious about how this may scale for a future state for when more blocks become available within the ["send a message" parent block](https://jetpackp2.wordpress.com/2020/06/25/whatsapp-send-a-message-block-master-thread/#comment-40358).

> I think it isn't a big deal. These blocks are special in the sense they can be usable with a free plan. Maybe we should reword the banner message?

Is it possible to have a different message within the Cover block or for any blocks that are usable on a Free plan but have some premium features? So for premium blocks "Upgrade your plan to use this premium block", and for blocks with premium features maybe something like "Upgrade your plan to use this premium feature"?

mreishus committed:
Allow "dual mode" (partially free) blocks to have customized banners

Current "Dual Mode" blocks:
- core/video: Allows users to upload images for free, but videos are premium.
- core/audio: Allows users to link to mp3s for free, but uploading one is premium.

The default message, "Upgrade your plan to use this premium block,"
doesn't let the user know that they can still use parts of the block for
free.

Here we have two customized messages:
core/video: 'Upgrade your plan to use video covers'
core/audio: 'Upgrade your plan to upload audio'

retrofox committed:
paid-blocks: do not extend icon free-usable blocks

retrofox commented:
> Yeah, maybe we could just remove the premium indicator for the Cover block? it would then alert users only when they try to use the premium feature. @davemart-in Do you have thoughts on that?

we decided to remove the icon for those blocks. Handled [here](https://github.com/Automattic/jetpack/pull/16611/commits/76600bcd80ca0d85a483d591ae051d2fe6cf3331)

mreishus commented:
Pushed a commit which allows for customized banners ( f9c4835 ).  I have two: 

-     core/video: 'Upgrade your plan to use video covers'
-     core/audio: 'Upgrade your plan to upload audio'

retrofox commented:
> Pushed a commit which allows for customized banners ( [f9c4835](https://github.com/Automattic/jetpack/commit/f9c48358a53d3b6f0f6755e16d279a36fa8b6aeb) ). I have two:
> 
> * core/video: 'Upgrade your plan to use video covers'
![image](https://user-images.githubusercontent.com/77539/90187792-c5b19480-dd90-11ea-9cdc-e89f50744910.png)

> * core/audio: 'Upgrade your plan to upload audio'

![image](https://user-images.githubusercontent.com/77539/90187825-d4984700-dd90-11ea-8cbf-7b923df4a1a9.png)

✅

mreishus committed:
Small cleanup

retrofox commented:
> I am however curious about how this may scale for a future state for when more blocks become available within the "send a message" parent block p1HpG7-9Af-p2 .

👀

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
Update extensions/extended-blocks/paid-blocks/README.md

Co-authored-by: Dave Smith <getdavemail@gmail.com>

retrofox committed:
use-upgrade-flow: first approach

retrofox committed:
upgrade-banner: use useUpgradeFlow() hook

mreishus committed:
Run prettier on files

retrofox committed:
upgrade-banner: fix getting upgrade data bug

retrofox committed:
paid-blocks: set required plan properly

mreishus committed:
use-upgrade-flow: Visual feedback to user that something is happening

retrofox commented on extensions/extended-blocks/paid-blocks/editor.scss:
```diff
@@ -40,6 +40,10 @@ $icon-background-color: #fff;
 			&:hover {
 				background: $studio-pink-30;
 			}
+			&:disabled {
+				opacity: 0.5;
+				cursor: wait;
+			}
```
Be careful about this, maybe it requires design feedback.

mreishus replied:
We're talking to Saul in slack

mreishus replied:
Updated here 3e9c5e6fe48f

mreishus committed:
use-upgrade-flow: Tweak visual feedback to user that something is happening

- Disabled -> .is-busy class for accessibility (Doesn't drop focus)
- Change button text
- Bring visual styles in line with autosave button in WordPress.org

mreishus commented:
Only 2 things stopping my :heavy_check_mark: :
1. Audio -> Upload mp3 -> Click upgrade -> Brings me to premium plan purchase, not personal plan.  (However, the donations block correctly goes to personal)
2. Payments and Premium Content blocks are showing no red circle in icon and old style upgrade buttons.  I feel like this is most likely me configuring the testing environment incorrectly, though.  Just reporting this in case. I applied two patches on my sandbox and synced built extensions from jetpack.
![2020-08-14_16-16](https://user-images.githubusercontent.com/937354/90293635-294bc880-de4a-11ea-9da3-a487007c95ba.png)
![2020-08-14_16-16_1](https://user-images.githubusercontent.com/937354/90293641-2c46b900-de4a-11ea-97bd-2d5534937fe2.png)

obenland committed:
Avoid "Invalid DOM property `stroke-width`" warning

obenland committed:
Avoid unique key warning

obenland committed:
Add dedicated function for plan retrieval

Separates update and plan checks for a clearer API.

retrofox commented:
> 1. Audio -> Upload mp3 -> Click upgrade -> Brings me to premium plan purchase, not personal plan.  (However, the donations block correctly goes to personal)

This something to handle in D46955-code

> Payments and Premium Content blocks are showing no red circle in icon and old style upgrade buttons. I feel like this is most likely me configuring the testing environment incorrectly, though. Reporting this just in case. I applied two patches on my sandbox and synced built extensions from jetpack.

Yes, it seems to be an issue compiling/syncing the code.

Thanks for your feedback.

mreishus commented:
Possible issue

- Have Personal plan on your blog (not premium)
- New post, Add Opentable block (requires premium)
- Press upgrade button inside block
  - It does not autosave (You get a popup asking you if you want to leave)
  - It does not do the animation

Is this because you're upgrading while paid and stuff is checking "isPaid" or something like that?

retrofox commented:
> getdave moved this from 👀 Ready for Review to 💻 In progress in Ajax Iteration Board 1 hour ago

@getdave, move it again to needs review since this is something that happens in prod. Guessing we can keep polishing it in another PR.

retrofox commented:
> Possible issue
> 
> * Have Personal plan on your blog (not premium)
> * New post, Add Opentable block (requires premium)
> * Press upgrade button inside block
>   
>   * It does not autosave (You get a popup asking you if you want to leave)
>   * It does not do the animation
> 
> Is this because you're upgrading while paid and stuff is checking "isPaid" or something like that?
> 
> Edit: I just did it again and it worked properly. When it failed I did this:
> New post -> Audio -> Upload mp3 -> Press button -> Buy Personal -> Redirect back (maybe it sent me to calypso?) -> Add opentable
> Edit 2: It's because when I purchase, it sends me back to calypso urls. It does work properly on wp-admins, but not the calypso (wordpress.com urls).

fixed here https://github.com/Automattic/jetpack/pull/16611/commits/ce1e6aacea263cf8c43b56cecb086c6386911da4

cc @mreishus

jeherve reviewed (changes requested):
This looks good to me. I still see the alignment issue with all blocks on WP 5.4 / no Gutenberg though:

<img width="807" alt="screenshot 2020-08-18 at 11 14 31" src="https://user-images.githubusercontent.com/426388/90496708-7d0a2a80-e146-11ea-8264-4b9644ddee6d.png">
<img width="829" alt="screenshot 2020-08-18 at 11 12 20" src="https://user-images.githubusercontent.com/426388/90496718-7f6c8480-e146-11ea-8f25-35c83745cee0.png">
<img width="794" alt="screenshot 2020-08-18 at 11 11 06" src="https://user-images.githubusercontent.com/426388/90496721-80051b00-e146-11ea-9267-f2184eaaba31.png">

This could be addressed in a follow-up PR though, I would not consider it a blocker.

However, there are some test failures with the new tests, that will block us from merging this PR:
https://travis-ci.com/github/Automattic/jetpack/jobs/372874966

```
FAIL extensions/extended-blocks/paid-blocks/test/render-premium-icon.js
  ● renderPaidIcon enhance the default block icon › when icon has src
    expect(received).toEqual(expected) // deep equality
    Expected: {"foreground": "#555d66", "src": <SVG xmlns="http://www.w3.org/2000/svg"><G /><_default /></SVG>}
    Received: serializes to the same string
      48 | describe( 'renderPaidIcon enhance the default block icon', () => {
      49 | 	it( 'when icon has src', () => {
    > 50 | 		expect( renderPaidIcon( iconWithSrc ) ).toEqual( iconWithSrcWithPaidIcon );
         | 		                                        ^
      51 | 	} );
      52 | 
      53 | 	it( 'when icon does not have src', () => {
      at Object.toEqual (extensions/extended-blocks/paid-blocks/test/render-premium-icon.js:50:43)
  ● renderPaidIcon enhance the default block icon › when icon does not have src
    expect(received).toEqual(expected) // deep equality
    Expected: <SVG xmlns="http://www.w3.org/2000/svg"><G /><_default /></SVG>
    Received: serializes to the same string
      52 | 
      53 | 	it( 'when icon does not have src', () => {
    > 54 | 		expect( renderPaidIcon( iconWithouthSrc ) ).toEqual( iconWithouthSrcWithPaidIcon );
         | 		                                            ^
      55 | 	} );
      56 | } );
      57 | 
      at Object.toEqual (extensions/extended-blocks/paid-blocks/test/render-premium-icon.js:54:47)
```

Commit:
Paid Blocks:
- adds key to PaidSymbol so tests pass
- renames file to render-paid-icon

retrofox commented:
thanks, @jeherve for your review. Tests are fixed [here](https://github.com/Automattic/jetpack/pull/16611/commits/2b8be42595c82734ac1da265e134e203a9960f9b):

![image](https://user-images.githubusercontent.com/77539/90501760-335c1880-e123-11ea-83a9-71e639a0d0dd.png)

It's in RfR again. 👍

retrofox committed:
upgrade-banner: add context property
Use this property to define the context where the banner is mounted. It could be used to add recording event properties, etc

retrofox committed:
upgrade-banner: tweak styles in editor canvas

retrofox committed:
paid-blocks: set banner context

retrofox commented:
> This looks good to me. I still see the alignment issue with all blocks on WP 5.4 / no Gutenberg th

Fixed.

jeherve reviewed (approved):
Looking good now. I haven't had the chance to test the WordPress.com part of this yet, but let's get this merged. We can always iterate in follow-up PRs if needed.

retrofox commented on extensions/blocks/videopress/editor.js:
```diff
@@ -18,7 +18,6 @@ import withVideoPressEdit from './edit';
 import withVideoPressSave from './save';
 import getJetpackExtensionAvailability from '../../shared/get-jetpack-extension-availability';
 import deprecatedV1 from './deprecated/v1';
-import wrapPaidBlock from '../../shared/wrap-paid-block';
```
Yes, let's create a janitorial issue since there other code that could be removed too

retrofox replied:
https://github.com/Automattic/jetpack/issues/16894

retrofox commented:
r212315-wpcom
